pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -105,27 +126,45 @@ func (bc *backendPoolTypeNodeIPConfig) CleanupVMSetFromBackendPoolByCondition(sl
 			}
 
 			newBackendPools[j] = bp
-			break
+			count++
```

Comment:
I think we can get rid of the break here as the length of backend pools is not supposed to be long.

Reply:
Updated.